erpnext/projects/doctype/project/project.py

```diff
@@ -325,11 +325,11 @@ def update_billed_amount(self):
 		# nosemgrep
 		total_billed_amount = frappe.db.sql(
 			"""select sum(base_net_amount)
-			from `tabSales Invoice Item` si_item, `tabSales Invoice` si
-			where si_item.parent = si.name
-				and if(si_item.project, si_item.project, si.project) = %s
-				and si.docstatus=1""",
-			self.name,
+			from `tabSales Invoice Item` si_item
+			join `tabSales Invoice` si on si_item.parent = si.name
+				where (si_item.project = %(name)s or (si_item.project is null and si.project = %(name)s))
```

Comment:
```suggestion
				where (si_item.project = %(name)s or (ifnull(si_item.project, '') = '' and si.project = %(name)s))
```

Empty string needs to be handled too. Recheck `explain` and `analyze` to ensure if perf fix is still valid.

Reply:
Mostly this ifnull won't cause any problem because MySQL will try to evaluate the second part of `and` condition using index. Though we never know for sure without running `analyze`.